lib/cmock_header_parser.rb

```diff
@@ -124,11 +168,20 @@ def import_source(source)
     src_lines = source.split(/\s*;\s*/).uniq
     src_lines.delete_if {|line| line.strip.length == 0}                            # remove blank lines
     src_lines.delete_if {|line| !(line =~ /[\w\s\*]+\(+\s*\*[\*\s]*[\w\s]+(?:\[[\w\s]*\]\s*)+\)+\s*\((?:[\w\s\*]*,?)*\s*\)/).nil?}     #remove function pointer arrays
-    if (@treat_externs == :include)
-      src_lines.delete_if {|line| !(line =~ /(?:^|\s+)(?:inline)\s+/).nil?}        # remove inline functions
+
+    unless (@treat_externs == :include)
```

Comment:
Hmmm... I'm not saying this section is wrong... It just seems a surprising change. Can you double check that this won't break externs?

Reply:
I'll review this section, it seems indeed that there is something missing. The extern stuff shouldn't be touched I guess, all inline stuff (if treat_inlines is enabled) should be gone before checking this. I'll get back to you with a proposal.

Reply:
I don't think this breaks externs. If I compare the old behaviour to the new behaviour:
**OLD BEHAVIOUR**

Option | :include | :exclude
---------|------------|-----------
:treat_extern | remove inline functions | remove extern + inline functions

**NEW BEHAVIOUR**

:treat_inline \ :treat_extern | :include | :exclude
---------------------------------|------------|-----------
**:include** | We remove _nothing_ | remove extern functions
**:exclude** | remove inline functions | remove extern + inline functions

So the old behaviour is preserved by specifying `:treat_inline: :exclude` (which is the default behaviour).

However, I do not take`:treat_extern` into account in the normalized_source, but I don't think we have to since this is just the header we use to compile the code, not to determine which functions to mock.

Does this seem correct or did I miss something?

Reply:
That sounds awesome. Thanks for double-checking!